### PR TITLE
Allow bools and dates as options for random_choice's short syntax

### PR DESCRIPTION
**Pull request by prescod**
random_choice has a short-form key:probability syntax. Now that short-form syntax can be used for booleans and dates, not just strings.

### CODE REVIEW

File: snowfakery/parse_recipe_yaml.py [hunk 1]
```diff
@@ -146,14 +146,34 @@ def removeline_numbers(dct: Dict) -> Dict:
     return {i: dct[i] for i in dct if i != "__line__"}
 
 
+def _coerce_to_string(val, context):
+    print(val, type(val))
```
Comment (boakley) [inline snowfakery/parse_recipe_yaml.py:150]: Was this print statement intentionally left in?

Comment (prescod) [reply snowfakery/parse_recipe_yaml.py]: Oops, no, fixed in a0fb7ef.

**Comment by prescod** [comment]
@boakley i fixed the issue you noticed, but it triggered a bug in `coverage`. The workaround is to improve coverage so I added an unrelated test to improve coverage.

**Review by boakley — COMMENTED** [review]
Looks good to me.

**Pull request merged**